statbot/emoji.py

```diff
@@ -0,0 +1,82 @@
+#
+# emoji.py
+#
+# statbot - Store Discord records for later analysis
+# Copyright (c) 2017 Ammon Smith
+#
+# statbot is available free of charge under the terms of the MIT
+# License. You are free to redistribute and/or modify it under those
+# terms. It is distributed in the hopes that it will be useful, but
+# WITHOUT ANY WARRANTY. See the LICENSE file for more details.
+#
+
+import unicodedata
+
+__all__ = [
+    'EmojiData',
+]
+
+class EmojiData:
+    __slots__ = (
+        'raw',
+        'id',
+        'unicode',
+        'custom',
+        'managed',
+        'name',
+        'category',
+        'roles',
+        'guild',
+    )
+
+    def __init__(self, emoji):
+        self.raw = emoji
+
+        if isinstance(emoji, str):
+            self.id = 0
+            self.unicode = emoji
+            self.custom = False
+            self.managed = False
+            self.name = [unicodedata.name(ch) for ch in emoji]
+            self.category = [unicodedata.category(ch) for ch in emoji]
+            self.roles = []
+            self.guild = None
+        else:
+            self.id = emoji.id
+            self.unicode = ''
+            self.custom = True
+            self.managed = getattr(emoji, 'managed', None)
+            self.name = [emoji.name]
+            self.category = ['custom']
+            self.roles = getattr(emoji, 'roles', None)
+            self.guild = getattr(emoji, 'guild', None)
+
+    @property
+    def mention(self):
+        if self.id:
+            return f'<:{self.name[0]}:{self.id}>'
+        else:
+            return self.unicode
```

Comment:
what is this function?

Reply:
It produces the "mention" for an emoji. Basically its raw representation in a string.

Reply:
is it used anywhere?

Reply:
No. It's sort of just meant as "completion" to what is basically the emoji class.

Reply:
I don't really understand why this is here. what's the downside to removing it?

Reply:
None right now, but it may be useful. Though it will probably end up being moved the client library.